.github/workflows/ci_pipeline.yml

```diff
@@ -65,7 +65,69 @@ jobs:
           echo "print('Loading sitecustomize.py...')
           import coverage
           coverage.process_startup() " > sitecustomize.py
-          coverage run -m pytest --continue-on-collection-errors -v --durations=0 pySDC/tests -m ${{ matrix.env }}
+          coverage run -m pytest --continue-on-collection-errors -v --durations=0 pySDC/tests pySDC/projects/*/tests -m ${{ matrix.env }}
+
+      - name: Make coverage report
+        run: |
+          mv data data_${{ matrix.python }}
+          coverage combine
+          mv .coverage coverage_${{ matrix.env }}_${{ matrix.python }}.dat
+
+      - name: Uploading artifacts
+        uses: actions/upload-artifact@v3
+        if: matrix.python == '3.10'
+        with:
+          name: cpu-test-artifacts
+          path: |
+            data_3.10
+            coverage_${{ matrix.env }}_3.10.dat
+
+  project_cpu_tests_linux:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python: ['3.8', '3.9', '3.10']
+        env: 
+          # - AllenCahn_Bayreuth
+          - AsympConv
+          # - compression
+          - DAE
+          - FastWaveSlowWave
+          # - GPU
+          - Hamiltonian
+          - matrixPFASST
+          - parallelSDC
+          - parallelSDC_reloaded
+          - PinTSimE
+          - RDC
+          - Resilience
+          # - SDC_showdown
+          - Second_orderSDC
+          - soft_failure
+          - TOMS
+
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Conda environment with Micromamba
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-file: "etc/environment-base.yml"
```

Comment:
This runs all of the project tests in the base environment except those with mpi4py, fenics or petsc, yes? I think what @pancetta wanted is to allow for project specific environments.
What about using [`conda env update`](https://docs.conda.io/projects/conda/en/latest/commands/update.html) with `--file` option? This way we could maybe define a base environment and build project environments on top which only need to specify what is different to the base environment.

Reply:
Added this now. It uses `micromamba install` because the `update` really only updates packages that are already installed.

So all projects now got an additional file `env_update.yml` where pytest is installed. Pytest also is included in the base-env, but the files cannot be empty. As an alterantive another package could also be used if preferred.